converter/internal/prometheusconvert/prometheusconvert_test.go

```diff
@@ -0,0 +1,114 @@
+package prometheusconvert_test
+
+import (
+	"bytes"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/grafana/agent/component/prometheus/remotewrite"
+	"github.com/grafana/agent/component/prometheus/scrape"
+	"github.com/grafana/agent/converter/internal/prometheusconvert"
+	"github.com/grafana/agent/pkg/flow"
+	"github.com/grafana/agent/pkg/river/ast"
+	"github.com/grafana/agent/pkg/river/vm"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	promSuffix = ".yaml"
+	flowSuffix = ".river"
+)
+
+func TestConvert(t *testing.T) {
+	filepath.WalkDir("testdata", func(path string, d fs.DirEntry, _ error) error {
+		if d.IsDir() {
+			return nil
+		}
+
+		if strings.HasSuffix(path, promSuffix) {
+			inputFile := path
+			expectFile := strings.TrimSuffix(path, promSuffix) + flowSuffix
+
+			inputBytes, err := os.ReadFile(inputFile)
+			require.NoError(t, err)
+			expectBytes, err := os.ReadFile(expectFile)
+			require.NoError(t, err)
+
+			caseName := filepath.Base(path)
+			caseName = strings.TrimSuffix(caseName, promSuffix)
+
+			t.Run(caseName, func(t *testing.T) {
+				testConverter(t, inputBytes, expectBytes)
+			})
+		}
+
+		return nil
+	})
+}
+
+func testConverter(t *testing.T, input, expect []byte) {
+	t.Helper()
+
+	actual, err := prometheusconvert.Convert(input)
+
+	require.NoError(t, err)
+	require.Equal(t, string(normalizeLineEndings(expect)), string(normalizeLineEndings(actual))+"\n")
+}
+
+// Replace '\r\n' with '\n'
+func normalizeLineEndings(data []byte) []byte {
+	normalized := bytes.ReplaceAll(data, []byte{'\r', '\n'}, []byte{'\n'})
+	return normalized
+}
+
+func TestFlowParsing(t *testing.T) {
+	filepath.WalkDir("testdata", func(path string, d fs.DirEntry, _ error) error {
+		if d.IsDir() {
+			return nil
+		}
+
+		if strings.HasSuffix(path, flowSuffix) {
+			caseName := filepath.Base(path)
+			caseName = strings.TrimSuffix(caseName, flowSuffix)
+
+			t.Run(caseName, func(t *testing.T) {
+				inputBytes, err := os.ReadFile(path)
+				require.NoError(t, err)
+
+				file, err := flow.ReadFile(path, inputBytes)
+				require.NoError(t, err)
+
+				if len(file.ConfigBlocks) > 0 {
+					require.FailNow(t, "This test doesn't support config blocks")
+				}
+
+				for _, node := range file.Components {
+					evalBlock(t, node)
+				}
+			})
+		}
+
+		return nil
+	})
+}
+
+func evalBlock(t *testing.T, block *ast.BlockStmt) {
```

Comment:
@rfratto  @mattdurham  This testing was updated to eval river blocks into their specific argument types. It's a little more maintenance which is why I wanted to avoid it, but evaluating the whole graph calls a whole ton of code in component New/Update functions. It was specifically crashing the test in windows but in general it's a lot of code I don't necessarily want to test here.

If there is some alternate way to do this without specifying the argument types based on the block name I am all ears.

Reply:
removed the test for now per a side conversation